### ISSUE TITLE
Provide means for resolvers to define their type name

### ISSUE THREAD

**Issue by konrad-garus** [graphql-java/graphql-java-tools#21]
Please consider adding means for resolvers to say what GraphQL type they resolve. It could be an annotation, or an overrideable method that defaults to class name. Either of these could be implement as backwards-compatible, non-breaking change.

Why?

I'm using graphql-java-tools with Spring. My root resolver happens to be a Spring bean that is using some AOP (e.g. for authorization). Spring does that by creating a CGLIB proxy, which is just a subclass. As a result, I get an instance of `Mutation$$EnhancerBySpringCGLIB$$2a4bb444` (subclass of `Mutation`). It blows up since the schema parser cannot find resolver for "Mutation".

I know I can configure schema parser using `dictionary`, but it's really tedious as I don't know the class type beforehand. It needs something like:

```
for (GraphQLResolver resolver : resolvers) {
    String name = resolver.getClass().getSimpleName();
    if (name.contains("CGLIB")) {
        name = resolver.getClass().getSuperclass().getSimpleName();
    }
    parser.dictionary(name, resolver.getClass());
}
```

This is really messy and tedious, compared to:

```
@Component
@GraphQLType("Mutation")
public class Mutation {
    // ...
}
```

or:

```
@Component
public class Mutation {
    // ...

    @Override
    public String getGraphQLType() {
        return "Mutation";
    }
}
```

**Comment by apottere** [comment]
I actually just released this in `3.1.0`, there's a `GraphQLQueryResolver`
and `GraphQLMutationResolver` interface.  Let me know if that works for you.

**Comment by konrad-garus** [comment]
That works like a charm, thank you!

I'm hitting another issue related to CGLIB though, will create new ticket.
